pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -190,6 +191,21 @@ func (r *Reconciler) create() error {
 		},
 	}
 
+	var userTags map[string]string
+	if r.featureGates.Enabled(configv1.FeatureGateGCPLabelsTags) {
+		userTags, err = util.GetResourceManagerTags(r.Context, r.coreClient, r.tagService, r.providerSpec.ResourceManagerTags)
+		if err != nil {
+			switch err.(type) {
+			case *machinecontroller.MachineError:
+				return err
+			}
+			return machinecontroller.CreateMachine("failed to fetch user-defined tags for %s: %v", r.machine.Name, err)
```

Comment:
This will be sufficient, if you using the wrapping semantic the core machine controller will pick up the invalid machine configuration, no need to use CreateMachine here
```suggestion
			return fmt.Errorf("failed to fetch user-defined tags for %s: %w", r.machine.Name, err)
```

Reply:
`MachineError` is used for just permission denied error case, wouldn't `CreateMachine` error be required for other errors where retry is required?

Reply:
Retry happens no matter if there's an error or not. I checked the main code and it doesn't handle the `CreateMachine` specially in this case, it just unwraps any errors to see if somewhere in the stack it was an `InvalidMachineConfiguration`, so the suggestion I've made should have the intended effect